webdriver/tests/bidi/browsing_context/close/prompt_unload.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+import asyncio
+
+from webdriver.bidi.modules.script import ContextTarget
+
+pytestmark = pytest.mark.asyncio
+
+CONTEXT_DESTROYED_EVENT = "browsingContext.contextDestroyed"
+USER_PROMPT_OPENED_EVENT = "browsingContext.userPromptOpened"
+
+
+@pytest.mark.parametrize("type_hint", ["window", "tab"])
+@pytest.mark.parametrize("prompt_unload", [None, False])
+async def test_prompt_unload_not_triggering_dialog(bidi_session, subscribe_events, url, type_hint, prompt_unload):
```

Comment:
Mind quickly explaining what the difference is between the two tests in this file? I have a problem to understand by reading the test name and going through the code, eg. why we only check the destroy event and remaining contexts in the second test.

Reply:
The idea behind the separation is as follows:
When you don't trigger `promptUnload=false` we want to make sure if the User Prompt event has not happened. We relay on the fact that `browsingContext.close` should be tested and close the context.

On the other hand, when we `promptUnload=true`, we also listen to the `browsingContext.destroyed` event to make sure the implementation doesn't only send the event without blocking the destruction.

Reply:
But shouldn't we ideally wait for the `browsingContext.contextDestroyed` event in both cases? It should always be emitted when a browsing context gets closed.

That raises another question: should we maybe move the checks for `browsingContext.contextDestroyed` to the `browsing_context/context_destroyed` folder where it IMO matches better? There we could have an extra test for the `promptUnload=true` case.

Reply:
For the first part - we have test verifying that the basecase is `browsingContext.contextDestroyed` and `browsingContext.close` success happen. I don't think we need to await everything everywhere. 

I don't think we should as we are not testing `browsing_context.contextDestroyed`. I think it should live under the parts being tested at for this is `browsingContext.close with promptUnload`.

Reply:
> For the first part - we have test verifying that the basecase is `browsingContext.contextDestroyed` and `browsingContext.close` success happen. I don't think we need to await everything everywhere.

Please note that we have a special case here. The `browsingContext.close` command will (most likely) return with `success` even through the browsing context hasn't been closed. Here it would be good to see that no `browsingContext.contextDestroyed` event has been send out.

I actually wonder if `browsingContext.close` should fail if the unload prompt gets dismissed.